monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/UploadNewPlugin.tsx

```diff
@@ -0,0 +1,185 @@
+import { useDropzone } from 'react-dropzone';
+import Alert from '@mui/material/Alert';
+import AlertTitle from '@mui/material/AlertTitle';
+import Button from '@mui/material/Button';
+import DeleteIcon from '@mui/icons-material/Delete';
+import FileUploadIcon from '@mui/icons-material/FileUpload';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import IslandHttpClient, { APIEndpoint } from '../../IslandHttpClient';
+import LoadingIcon from '../LoadingIcon';
+import '../../../styles/components/plugins-marketplace/UploadNewPlugin.scss';
+
+const getColor = (props) => {
+  if (props.isDragAccept) {
+    return '#00e676';
+  }
+  if (props.isDragReject) {
+    return '#ff1744';
+  }
+  if (props.isFocused) {
+    return '#ffc107';
+  }
+  return '#eeeeee';
+};
+
+const UploadNewPlugin = () => {
+  const [plugin, setPlugin] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [showSuccessAlert, setShowSuccessAlert] = useState(false);
+  const [showErrorAlert, setShowErrorAlert] = useState(false);
+  const [pluginName, setPluginName] = useState('');
+  const [errors, setErrors] = useState([]);
+
+  const onDrop = useCallback((acceptedPlugin, rejectedPlugin) => {
+    if (acceptedPlugin?.length) {
+      const reader = new FileReader();
+      reader.onload = (e) => {
+        if (e.target.readyState === FileReader.DONE) {
+          const binaryPlugin = new Uint8Array(e.target.result);
+          setPlugin(binaryPlugin);
+          setPluginName(Object.assign(acceptedPlugin?.[0]).name);
+        }
+      };
+      reader.readAsArrayBuffer(acceptedPlugin?.[0]);
+    }
+    if (rejectedPlugin?.length) {
+      const uniqueErrors = new Set();
+
+      rejectedPlugin.forEach(item => {
+        item.errors.forEach(error => {
+          uniqueErrors.add(`${error.message}`);
+        });
+      });
+      setErrors(Array.from(uniqueErrors));
+      setShowErrorAlert(true);
+      setTimeout(() => {
+        setShowErrorAlert(false);
+        setErrors([]);
+      }, 10000);
+    }
+  }, []);
+
+  const {
+    getRootProps,
+    getInputProps,
+    isDragAccept,
+    isFocused,
+    isDragReject
+  } = useDropzone({
+    accept: {
+      'application/x-tar': []
+    },
+    maxFiles: 1,
+    onDrop
+  });
+
+  const uploadPlugin = () => {
+    setLoading(true);
+    IslandHttpClient.put(APIEndpoint.installAgentPlugin, plugin, false).then(res => {
+      if (res.status === 200) {
+        setLoading(false);
+        setShowSuccessAlert(true);
+        setTimeout(() => {
+          setShowSuccessAlert(false);
+          setPluginName('');
+        }, 10000);
+        setPlugin(null);
+      } else {
+        let error = `Error occurred installing the plugin archive '${pluginName}'`;
+        setErrors(prevErrs => [...prevErrs, error]);
+        setPlugin(null);
+        setPluginName('');
+        setLoading(false);
+        setShowErrorAlert(true);
```

Comment:
It looks like this "show error" code is used elsewhere. We could extract it into a function